packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/spot-event-plugin-client.ts

```diff
@@ -94,6 +109,49 @@ export class SpotEventPluginClient {
     }
   }
 
+  public async getGroupPoolCollection(isGroup: boolean): Promise<string[]> {
```

Comment:
Pools and groups are very similar in Deadline, they even use the same object type `PoolCollections`.
I can't see any reasons to duplicate code for these two instances.

Reply:
I think we can take this opportunity to establish a pattern in our code that wraps the `DeadlineClient.GetRequest/PostRequest` methods. While groups and pools may be (coincidentally) returning the same data structure, from an API perspective they are different resources, so I think we should reflect that in our interfaces.

To me, this code expresses its intent a bit better
```ts
const pools = await client.getPools();
pools.push('my_new_pool');
await client.postPools(pools);
```
than
```ts
const pools = await client.getGroupPoolCollection(false);
pools.push('my_new_pool');

// accidentally mixed up the boolean here, so this will overwrite the users groups with these pools
await client.saveGroupPoolCollection(pools, true);
```

I think the benefits of having a more intuitive interface outweigh having to split these methods up with some  code duplication. What do you think?